src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (Array.isArray(this) && this.length > 1) {
+      let str;
```

Comment:
```suggestion
      let str = '';
```

Reply:
I made "str" as a string when added its first element as a string.

Reply:
Yes, and I wrote that you don't need to duplicate this condition. You can check this element in a loop too, and make str = '' initially.